modules/ding_webtrends/ding_webtrends.module

```diff
@@ -151,17 +151,34 @@ function ding_webtrends_get_search_info() {
 
 /**
  * Implements hook_preprocess_ding_list_add_button().
+ */
+function ding_webtrends_preprocess_ding_list_add_button(&$variables) {
+  $variables = _ding_webtrends_preprocess_ding_list_add_button($variables);
+
+}
+
+/**
+ * Implements hook_preprocess_ding_list_add_button_simple().
+ */
+function ding_webtrends_preprocess_ding_list_add_button_simple(&$variables) {
+  $variables = _ding_webtrends_preprocess_ding_list_add_button($variables);
+}
+
+/**
+ * Helper function for preprocessing ding list add button.
  *
  * The ding list add buttons is AJAX-enabled and we want to track those with
  * dcsMultiTrack. Unfortunately, the standard classes is not enough to make the
  * selector specific enough, so we add our own here.
  */
-function ding_webtrends_preprocess_ding_list_add_button(&$variables) {
+function _ding_webtrends_preprocess_ding_list_add_button($variables) {
```

Comment:
Why not parse the $variables as ref. the whole way around? It kind of jump into my eyes that the $variables where total overriden in the functions calling this function.

Drupal normally change the $vars variable not override it. I known the end result is the same.

Reply:
Maybe also remove the word "preprocess" from the helper function as that makes it look like a hook function. It might lead to confusion later on.

Reply:
In this case, we want to mutate/extend the $variables, which is why i think the $var = $newvar is a better syntax for readability

Just using some_function($var) signals in my brain that literally anything could be going on - maybe we are mutating a third part based on $var..

Does that make sense? :)

Reply:
Also, "preprocess" has been replaced by "prepare" in Fixed in 440b09f 👍